### DISCUSSION TITLE
subdomain

### DISCUSSION THREAD

**Discussion by MamadouSDiallo** [litestar-org/litestar#1684] [Q&A] [answered]
Hi all 

I would like to know if litestar support subdomain e.g. ```subdomain.example.com```. 

Below I used chatGPT to general an minimumal example with starlette.

```py
from starlette.applications import Starlette
from starlette.routing import Route, Mount

async def index(request):
    return PlainTextResponse('Welcome to the main domain!')

async def subdomain(request):
    return PlainTextResponse('Welcome to the subdomain!')

routes = [
    Route('/', endpoint=index),
    Route('/', endpoint=subdomain, methods=["GET"], host="subdomain.example.com"),
]

app = Starlette(routes=routes)

if __name__ == '__main__':
    import uvicorn
    uvicorn.run(app, host='0.0.0.0', port=8000)
```

**Comment by JacobCoffee** [comment, marked as answer]
Hi, we do not support host-based routing.

https://docs.litestar.dev/2/migration/fastapi.html#host-based-routing

> Host based routing
Host based routing class is intentionally unsupported. If your application relies on Host you will have to separate the logic into different services and handle this part of request dispatching with a proxy server like [nginx](https://www.nginx.com/) or [traefik](https://traefik.io/).
